core/trino-main/src/main/java/io/trino/operator/scalar/SequenceFunction.java

```diff
@@ -132,10 +161,10 @@ public static void checkValidStep(long start, long stop, long step)
                 "sequence stop value should be greater than or equal to start value if step is greater than zero otherwise stop should be less than or equal to start");
     }
 
-    public static void checkMaxEntry(int length)
+    public static void checkMaxEntry(long length)
```

Comment:
Consider returning the int value from this method. This method is always followed by `toIntExact()`.